src/client/mvt/feature-decoder.js

```diff
@@ -130,15 +162,102 @@ function _getPreClippedVertices(geom, mvtExtent) {
         let x = coord.x;
         let y = coord.y;
 
-        x = 2 * x / mvtExtent - 1;	
-        y = 2 * (1 - y / mvtExtent) - 1;	
+        x = 2 * x / mvtExtent - 1;
+        y = 2 * (1 - y / mvtExtent) - 1;
 
         return [x, y];
     });
 }
 
+function clipLine(line) {
+    // linestring clipping based on the Cohen-Sutherland algorithm
+    // input is a single linestring [point0, point1, ...]
+    // output is an array of flat linestrings:
+    // [[p0x, p0y, p1x, p1y, ...], ...]
+    let clippedLine = [];
+    const clippedLines = [];
+    function clipType(point) {
+        let type = 0;
+        for (let i = 0; i < 4; i++) {
```

Comment:
Why 4?

Reply:
These are the edges 0=right, 1=top, 2=left, 3=bottom as defined by `clippingEdges` and `clippingEdgeIntersectFn`.

Reply:
In this case, I would define the different edges to understand this:

`const RIGHT_EDGE = 0`
`const TOP_EDGE = 1`
`const LEFT_EDGE = 2`
`const BOTTOM_EDGE = 3`

Reply:
You're removing all the mystery from the classic geometry algorythms :joy: